src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/EventBookingManager.java

```diff
@@ -394,6 +444,104 @@ public EventBookingDTO requestBooking(final IsaacEventPageDTO event, final Regis
         }
     }
 
+    /**
+     * Request a reservation for the given batch of users on the given event.
+     * This method will allow teachers and group managers to "soft-book" students providing they do not exceed their
+     * allocated limit.
+     * There is no additional event information passed now, this can be filled in later when actual bookings are
+     * requested.
+     *
+     * @param event - to reserve the user on
+     * @param users - to reserve on the event
+     * @return confirmation of reservation
+     */
+    public List<EventBookingDTO> requestReservations(final IsaacEventPageDTO event, final List<RegisteredUserDTO> users,
+                                                     final RegisteredUserDTO reservingUser)
+            throws EventDeadlineException, EmailMustBeVerifiedException, DuplicateBookingException,
+            SegueDatabaseException, EventIsFullException, EventGroupReservationLimitException {
+
+        // af599 TODO: Is it wise to do this before acquiring a database lock?
+        for (RegisteredUserDTO user : users) {
+            this.ensureValidBooking(event, user, true, BookingStatus.RESERVED);
+        }
+
```

Comment:
We should answer this. From the looks of things isUserBooked(...) and isUserReserved(...) check the database event booking records for that user. One possible problematic execution trace would be if the student completes a booking between the time that the reserving teacher does this check and then reserves that student. That would create two bookings which count towards capacity for the same student.

Reply:
Fixed in e251ca97.